rust/lance-index/src/vector/pq/storage.rs

```diff
@@ -560,7 +554,7 @@ impl PQDistCalculator {
         let distance_table = if matches!(distance_type, DistanceType::Cosine | DistanceType::L2) {
             build_distance_table_l2(codebook, num_bits, num_sub_vectors, query)
         } else {
-            unimplemented!("DistanceType is not supported: {:?}", distance_type);
+            build_distance_table_dot(codebook, num_bits, num_sub_vectors, query)
```

Comment:
Can we make this a normal `match` statement instead of `matches`?

```
let distance_table = match distance_type {
  DistanceType::Cosine | DistanceType::L2 => ...
  DistanceType::Dot => ...
};
```
This way, if we introduce a new distance type in the future, it won't silently start using `dot`.

Reply:
Fixed